keps/6757-failure-recovery/README.md

```diff
@@ -0,0 +1,379 @@
+    gracePeriodLeft := pod.DeletionTimestamp.Add(totalGracePeriod).Sub(now)
+    return ctrl.Result{RequeueAfter: gracePeriodLeft}, nil
+  }
+
+  // ...
+}
+```
+
+In that scheduled reconciliation, unless the node recovered or the pod was deleted,
+the pod will be deemed "zombie" and transitioned into the `PodFailed` phase:
+
+```go
+func (r *ZombiePodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+  // ...
+
+  pod.Status.Phase = corev1.PodFailed
+  if err := r.client.Status().Update(ctx, pod); err != nil {
+    return ctrl.Result{}, err
+  }
+
+  return ctrl.Result{}, nil
+}
+```
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+#### Unit Tests
+
+The proposal will be covered with unit tests for:
+1. Configuration parsing.
+1. Controller behavior:
+    1. Whether it ignores irrelevant pods (not terminating, already failed/succeeded, not annotated).
+    1. Whether it correctly schedules a reconciliation for when the grace period elapses.
+    1. Whether it updates the pod's phase to `Failed` after the grace period elapses.
+
+#### Integration Tests
+
+The proposal will be covered with integrations tests that check whether:
+1. Adding a deletion timestamp to the pod requeues a reconciliation loop for when the grace period elapses.
+1. After the grace period elapses, the pod is marked as `Failed`.
+1. Replacement pods are scheduled in the place of the failed pod (optional, technically 1+2 is sufficient to prove this).
+
+Existing integration tests should prove that this feature does not impact Kueue during normal operation.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature gate disabled by default.
+- Positive feedback from the users.
+
+### Beta
+
+- Feature gate enabled by default.
+- Re-evaluate the introduction of the `FailureRecoveryPolicy` API.
+
+## Implementation History
+
+2025-09-08: The [issue](https://github.com/kubernetes-sigs/kueue/issues/6757) is raised in Kueue.
+
+2025-09-12: The [issue](https://github.com/kubernetes/kubernetes/issues/134038) is raised in core Kubernetes.
+
+2025-10-17: First draft of the KEP.
+
+## Drawbacks
+
+* The same feature is discussed in core Kubernetes ([kubernetes/issues/134038](https://github.com/kubernetes/kubernetes/issues/134038)), so the underlying issue could potentially be fixed upstream. The timeline of an upstream change is long, but if the feature is deemed not time-critical, it could be fixed at the source instead of in Kueue.
+* This feature introduces a potential footgun to Kueue users and could turn out to be to volatile/risky to use in most affected cases.
+* It introduces yet another responsibility for Kueue - on top of quota management and scheduling, it will also start performing failure recovery.
+
+## Alternatives
```

Comment:
IIUC, Non Graceful Shutdown (https://kubernetes.io/docs/concepts/cluster-administration/node-shutdown/#non-graceful-node-shutdown) could be considered as an alternative.